### DISCUSSION TITLE
Can I use imgproxy with cloudflare R2 as client

### DISCUSSION THREAD

**Discussion by omnaiduu** [imgproxy/imgproxy#1234] [General]
I am not able to configure cloudflare R2 which has S3 compatible API with imgproxy.. I find the key is not found and 404.. is it even possible to connect.. can anyone please help me with this.. there's no documentation also

**Comment by DarthSim** [comment]
Hi @cloudcreatr!

Sure you can use R2 with imgproxy. First, you need to create an API token for R2 in Cloudflare's dashboard. This will give you an S3 API access key ID, an S3 API secret access key, and an S3 API endpoint.

Then, just configure your imgproxy like this:
```
AWS_REGION=auto
AWS_ACCESS_KEY_ID=<your access key ID>
AWS_SECRET_ACCESS_KEY=<secret access key>
IMGPROXY_S3_ENDPOINT=<your S3 API endpoint>
IMGPROXY_S3_MULTI_REGION=0
```

... and use `s3://<your-bucket-name>/<object-path>` as a source URL

**Reply by omnaiduu** [reply]
![Screenshot_2023-12-22-01-31-25-156_com android chrome](https://github.com/imgproxy/imgproxy/assets/79000269/1cd5a244-67fa-4de6-a714-71eed16eafed)
https://img.cloudcreatr.com/preset:sharp/resize:fill:300:400:0/gravity:sm/plain/s3://im-resize/myimg.jpg@webp
![Screenshot_2023-12-22-01-33-59-564_com android chrome](https://github.com/imgproxy/imgproxy/assets/79000269/2e13e183-9cd2-4ad4-a914-40863c172611)
Its not working..

**Reply by omnaiduu** [reply]
It's working.. mistake was in endpoint url.. i copied it directly from dashboard which has the bucket name behind it..
https://{someiddd}.r2.cloudflarestorage.com/{buckedname}❌
https://{someiddd}.r2.cloudflarestorage.com/✅
